### ISSUE TITLE
Unstable travis-ci builds

### ISSUE THREAD

**Issue by ZauberNerd** [reactjs/react-docgen#110]
It seems like there are a lot of random build failures for node v4 where the
worker-farm used by jest is being terminated.
After reading a couple of issues it seems to be related to
[jest consuming a lot of memory](https://github.com/facebook/jest/issues/815).

In the past I've also experienced strange behaviour related to memory consumption
on travis-ci (although my issues were with gradle for react-native).
But I remembered that I worked around these issues by forcing travis to use a
non-docker worker to run the build.
So I did a couple of test runs for react-docgen with and without travis' docker
containers: https://travis-ci.org/ZauberNerd/react-docgen/builds
And indeed it seems like the builds are running stable on non-docker containers.

This is by far not a bullet proof analysis on what and why is happening with
travis, jest, etc. But on docker containers there are 6/10 successful builds
and on the ubuntu boxes it is 10/10 successful builds.

In conclusion: It might be worth considering to use ubuntu boxes on travis-ci
to have a stable build environment to be able to accept and merge PRs with
confidence.
Other solutions could be to wait for / submit patches to jest / travis or
switching to a different test framework (ava worked pretty well for me) or
switching to a different CI provider.

The changes required are: [setting `sudo: required` in the travis.yml.](https://github.com/ZauberNerd/react-docgen/commit/66f69d951b39a35f2a2b5945c71919f30951fe74)

See also: [travis-ci documentation on build environments](https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments)

**Comment by fkling** [comment]
Thank you so much for looking into this. We can definitely set `sudo: required` and rerun the checks on all outstanding PRs.
